pkg_config.bzl

```diff
@@ -86,15 +88,28 @@ def _ignore_opts(opts, ignore_opts):
             remain += [opt]
     return remain
 
+def _symlink_no_duplicate(ctx, src, dest):
+    if not dest.exists:
+        ctx.symlink(src, dest)
+        return True
+    return False
+
 def _symlinks(ctx, basename, srcpaths):
     result = []
     root = ctx.path("")
     base = root.get_child(basename)
     rootlen = len(str(base)) - len(basename)
     for src in [ctx.path(p) for p in srcpaths]:
         dest = base.get_child(src.basename)
-        ctx.symlink(src, dest)
-        result += [str(dest)[rootlen:]]
+        # if the directory already exists then:
+        # 1) attempt to modify the name with the parent directory name
+        # 2) if that fails don't create the symlink at all
+        # we can sometimes skip symlink creation because of ocassional
+        # duplicates in pkg_config output
+        if dest.exists:
+            dest = base.get_child(src.basename + src.dirname.basename)
+        if _symlink_no_duplicate(ctx, src, dest):
+            result += [str(dest)[rootlen:]]
```

Comment:
Modifying the name of the symlink that is created like this may lead to nondeterministic include path naming (or spooky action at a distance, where include path names depend on the order in which symlinks are created). Moreover, if you look at how the return value of `_symlinks` is used, you should see that this changes how the `strip_include` option to the `pkg_config` rule works.

For example, consider a package with include paths `/usr/include/x86_64-linux-gnu/MyPackage` and `/usr/include/MyPackage`. It provides the include files `/usr/include/x86_64-linux-gnu/MyPackage/my.h` and `/usr/include/MyPackage/package.h`.

A user of the `pkg_config` Bazel rule might specify a `strip_prefix` value of `MyPackage`, so that they can just `#include "my.h"` and `#include "package.h"` directly. 

Under this PR, Bazel will create symlinks named `includes/MyPackage` and `includes/include/MyPackage` or `includes/x86_64-linux-gnu/MyPackage`, depending on which gets created first. There is no single `strip_prefix` value that can correctly strip out both of these. Furthermore, the names of the symlinks that get created depends on the order in which they are created, which is undesirable for multiple reasons.

Reply:
yes, that makes a lot of sense; a `strip_prefix` or `prefix` (for adding a path prefix)